integration/presto/src/main/prestosql/org/apache/carbondata/presto/PrestoFilterUtil.java

```diff
@@ -0,0 +1,330 @@
+          // there can be multiple ranges for a single column. combined to create colValueExpression
+          Expression rangeExpression = null;
+          if (!range.getLow().isLowerUnbounded()) {
+            Object value = convertDataByType(range.getLow().getValue(), type);
+            switch (range.getLow().getBound()) {
+              case ABOVE:
+                rangeExpression =
+                    new GreaterThanExpression(colExpression, new LiteralExpression(value, coltype));
+                break;
+              case EXACTLY:
+                rangeExpression = new GreaterThanEqualToExpression(colExpression,
+                    new LiteralExpression(value, coltype));
+                break;
+              case BELOW:
+                throw new IllegalArgumentException("Low marker should never use BELOW bound");
+              default:
+                throw new AssertionError("Unhandled bound: " + range.getLow().getBound());
+            }
+          }
+
+          if (!range.getHigh().isUpperUnbounded()) {
+            Expression lessThanExpression;
+            Object value = convertDataByType(range.getHigh().getValue(), type);
+            switch (range.getHigh().getBound()) {
+              case ABOVE:
+                throw new IllegalArgumentException("High marker should never use ABOVE bound");
+              case EXACTLY:
+                lessThanExpression = new LessThanEqualToExpression(colExpression,
+                    new LiteralExpression(value, coltype));
+                break;
+              case BELOW:
+                lessThanExpression =
+                    new LessThanExpression(colExpression, new LiteralExpression(value, coltype));
+                break;
+              default:
+                throw new AssertionError("Unhandled bound: " + range.getHigh().getBound());
+            }
+            rangeExpression = (rangeExpression == null ?
+                lessThanExpression :
+                new AndExpression(rangeExpression, lessThanExpression));
+          }
+          colValueExpression = (colValueExpression == null ?
+              rangeExpression :
+              new OrExpression(colValueExpression, rangeExpression));
+        }
+      }
+
+      if (singleValues.size() == 1) {
+        colValueExpression = new EqualToExpression(colExpression,
+            new LiteralExpression(singleValues.get(0), coltype));
+      } else if (singleValues.size() > 1) {
+        List<Expression> exs =
+            singleValues.stream().map((a) -> new LiteralExpression(a, coltype)).collect(toList());
+        colValueExpression = new InExpression(colExpression, new ListExpression(exs));
+      }
+
+      if (colValueExpression != null) {
+        finalFilters = (finalFilters == null ?
+            colValueExpression :
+            new AndExpression(finalFilters, colValueExpression));
+      }
+    }
+    return finalFilters;
+  }
+
+  private static Object convertDataByType(Object rawdata, HiveType type) {
```

Comment:
```suggestion
  private static Object convertDataByType(Object rawData, HiveType type) {
```

Reply:
ok. These are from base code, I will cross check once in all the places and handle in this PR

Reply:
ok, checked similar issues and fixed also